sqlglot/tokens.py

```diff
@@ -851,3 +917,24 @@ def _scan_var(self):
             else:
                 break
         self._add(self.KEYWORDS.get(self._text.upper(), TokenType.VAR))
+
+    def _extract_string(self, delimiter):
```

Comment:
this looks similar to _scan_string, again any opp to dry this up?

Reply:
Yes, there is probably some opportunity to use a shared helper method.

Reply:
refactored